src/java/org/apache/cassandra/io/tries/Walker.java

```diff
@@ -83,9 +83,9 @@ protected final void go(long position)
         long curOffset = position - bh.offset();
         if (curOffset < 0 || curOffset >= buf.limit())
         {
-            BufferHolder currentBh = bh;
+            bh.release();
+            bh = Rebufferer.EMPTY;
```

Comment:
I believe this needs a comment (e.g. `// prevent double release if the call below throws`).

`RandomAccessReader.reBufferAt` also needs this line between `release` and `rebuffer`.

Reply:
ok